### DISCUSSION TITLE
avocado.test.progress logging stream setup via API

### DISCUSSION THREAD

**Discussion by marioaag** [avocado-framework/avocado#5304] [Q&A] [answered]
I'm trying to send some progress messages to the terminal using the documentation in: https://avocado-framework.readthedocs.io/en/92.0/guides/writer/chapters/logging.html#showing-custom-log-streams

And setting the core.show through the job_config parameters

I can see how it gets set in the job.log file:
```
2022-03-31 16:36:18,580 job              L0283 I    | Avocado version: 92.0
2022-03-31 16:36:18,580 job              L0284 I    |
2022-03-31 16:36:18,580 job              L0262 I    | Avocado config:
2022-03-31 16:36:18,580 job              L0263 I    |
2022-03-31 16:36:18,582 job              L0265 I    | {'core.input_encoding': 'utf-8',
2022-03-31 16:36:18,582 job              L0265 I    |  'core.show': ['app', 'avocado.test.progress'],      <<<<<------ GETS SET
2022-03-31 16:36:18,582 job              L0265 I    |  'datadir.paths.base_dir': '/home/mario/avocado',
2022-03-31 16:36:18,582 job              L0265 I    |  'datadir.paths.cache_dirs': ['/home/mario/avocado/data/cache'],
2022-03-31 16:36:18,582 job              L0265 I    |  'datadir.paths.data_dir': '/home/mario/avocado/data',
2022-03-31 16:36:18,582 job              L0265 I    |  'datadir.paths.logs_dir': '/tmp/avocado_logs',
2022-03-31 16:36:18,582 job              L0265 I    |  'datadir.paths.test_dir': '/usr/share/doc/avocado/tests',
2022-03-31 16:36:18,582 job              L0265 I    |  'human_ui.omit.statuses': [],
```

And I can see how the message is logged in the debug.log
```
[stdlog] 2022-03-31 16:36:22,425 ErrorInjects     L0069 INFO | 1-ErrorInjects.node_remove_replace: Running Inject 1
```

But not see it in the terminal output:
```
JOB ID     : 0dd62b88a307c200110f7a02481a0d2ef827ec5c
JOB LOG    : /tmp/avocado_logs/job-2022-03-31T16.36-0dd62b8/job.log
 (220331-143618368415.CMD.-1/1) /home/mario/git/Gatekeepers/test/avocado/testcases/ErrorInjects/ErrorInjects.py:ErrorInjects.node_remove_replace: STARTED
 (220331-143618368415.CMD.-1/1) /home/mario/git/Gatekeepers/test/avocado/testcases/ErrorInjects/ErrorInjects.py:ErrorInjects.node_remove_replace: PASS (12.95 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /tmp/avocado_logs/job-2022-03-31T16.36-0dd62b8/results.html
JOB TIME   : 14.34 s
```

In the test what I'm doing is (avocado instrumented):
```
inject_number = 1
progress_log = logging.getLogger("avocado.test.progress")
progress_log.info(f'{self.name}: Running Inject {inject_number}')
sleep(10)
``

Do I missing something else that should be done to get it printing to the terminal?

**Comment by richtja** [comment]
Hi @marioaag thank you for reporting this. This is definitely a bug, and I created issue #5305 to track it. For now, you can move 'core.show' option into the test suite config with that, it is working fine.

```
#!/usr/bin/env python3

import sys

from avocado.core.job import Job
from avocado.core.suite import TestSuite

config = {'resolver.references': ['/tmp/test.py:PassTest.test'], 'core.show': ['app', 'avocado.test.progress']}

suite = TestSuite.from_config(config)
with Job(config, [suite]) as j:
    sys.exit(j.run())
```

**Reply by marioaag** [reply]
Right now I'm using TestSuite() constructor due I need to provide specific parameters to each one of the runnables, so couple of questions whit this.
1) Is it possible to provide run.test_parameters to each one of the references using from_config() constructor?

2) In the scenario where I'm seeing the error I'm already passing 'core.show': ['app', 'avocado.test.progress'] to the TestSuite() constructor to job_config and config parameters and also passing it to Job() config parameter, it was missing in Runnable() constructor so I added it as part of the config parameter and still not getting the expected message, is a way to workaround the issue using TestSuite() constructor?

3) I'm running in 9.2 version, does that make a difference? I didn't update to the latest due some changes in the internal core API broken my actual implementation

**Reply by richtja** [reply]
>     3. I'm running in 9.2 version, does that make a difference? I didn't update to the latest due some changes in the internal core API broken my actual implementation

IIUIC you meant avocado version 92. This might be the problem because the `show` option was fixed in 93. version in #5056. Since the 92 is the LTS version, I think we should fix this there also. @clebergnu @beraldoleal @ana what do you think?

**Comment by richtja** [comment, marked as answer]
Hi @marioaag  you can see the fix in #5306, but for combining the job config and suite config you have to use `job_config` argument in `Suite.from_config`

**Reply by marioaag** [reply]
Thanks for the info @richtja, I already validate that it works on 94.0 and 96.0 and master!
